driver-core/src/main/com/mongodb/internal/operation/OperationHelper.java

```diff
@@ -174,10 +205,42 @@ static void validateWriteRequestCollations(final ConnectionDescription connectio
         validateCollationAndWriteConcern(connectionDescription, collation, writeConcern);
     }
 
+    static void validateUpdateRequestArrayFilters(final ConnectionDescription connectionDescription,
+                                                  final List<? extends WriteRequest> requests, final WriteConcern writeConcern) {
+        List<BsonDocument> arrayFilters = null;
+        for (WriteRequest request : requests) {
+            if (request instanceof UpdateRequest) {
+                arrayFilters = ((UpdateRequest) request).getArrayFilters();
+            }
+            if (arrayFilters != null) {
+                break;
+            }
+        }
+        validateArrayFiltersAndWriteConcern(connectionDescription, arrayFilters, writeConcern);
```

Comment:
Let's simplify this code a bit: move this line right before the break statement above, and move the arrayFilters variable into the for loops.  Then you know that arrayFilters is not null, and you don't have to compare arrayFilters to null in validateArrayFiltersAndWriteConcern.

Same for the validateWriteRequestHints below